src/wasm/wasm-validator.cpp

```diff
@@ -2318,10 +2318,12 @@ void FunctionValidator::visitStructNew(StructNew* curr) {
                       "struct.new must have the right number of operands")) {
       // All the fields must have the proper type.
       for (Index i = 0; i < fields.size(); i++) {
-        shouldBeSubType(curr->operands[i]->type,
-                        fields[i].type,
-                        curr,
-                        "struct.new operand must have proper type");
+        shouldBeSubType(
+          curr->operands[i]->type,
+          fields[i].type,
+          curr,
+          ("struct.new operand " + std::to_string(i) + " must have proper type")
+            .c_str());
```

Comment:
I wonder if this is not undefined behavior. You are creating a temporary `std::string`, then getting a C string from it, but the `std::string` may not live long enough for the call to occur. Can try to revert this to see if it fixes the windows bot...